internal/generator/vector/output/azuremonitor/azuremonitor.go

```diff
@@ -45,43 +43,33 @@ shared_key = "{{.SharedKey}}"
 {{end}}`
 }
 
-func New(id string, o logging.OutputSpec, inputs []string, secret *corev1.Secret, strategy common.ConfigStrategy, op framework.Options) []framework.Element {
+func New(id string, o obs.OutputSpec, inputs []string, secrets vectorhelpers.Secrets, strategy common.ConfigStrategy, op framework.Options) []framework.Element {
 	dedottedID := vectorhelpers.MakeID(id, "dedot")
 	if genhelper.IsDebugOutput(op) {
 		return []framework.Element{
 			Debug(vectorhelpers.MakeID(id, "debug"), vectorhelpers.MakeInputs(inputs...)),
 		}
 	}
-	return framework.MergeElements(
-		[]framework.Element{
-			viaq.DedotLabels(dedottedID, inputs),
-			Output(id, o, []string{dedottedID}, secret, op),
-			common.NewAcknowledgments(id, strategy),
-			common.NewBatch(id, strategy),
-			common.NewBuffer(id, strategy),
-			common.NewRequest(id, strategy),
-		},
-		TLSConf(id, o, secret, op),
-	)
-}
-
-func Output(id string, o logging.OutputSpec, inputs []string, secret *corev1.Secret, op framework.Options) framework.Element {
 	azm := o.AzureMonitor
-	return AzureMonitor{
+	e := AzureMonitor{
```

Comment:
nit:   more explicit name
```suggestion
	output := AzureMonitor{
```

Reply:
This was part of the azure PR that already merged